ui/src/app/components/runtime-configuration-panel.tsx

```diff
@@ -582,13 +584,30 @@ const PanelMain = fp.flow(
             () => {
               if (attachedPdExists) {
                 return (
-                  <ConfirmDeleteRuntimeWithPD
-                    onConfirm={async (runtimeStatusReq) => {
+                  <ConfirmDeleteEnvironmentWithPD
+                    onConfirm={async (deletePDSelected) => {
+                      const runtimeStatusReq = switchCase(
```

Comment:
This logic was moved from the `ConfirmDeleteEnvironmentWithPD` component to make it environment-agnostic.